src/inner_product_proof.rs

```diff
@@ -400,10 +400,10 @@ impl InnerProductProof {
         }
 
         let pos = 2 * lg_n * 32;
-        let a =
-            Scalar::from_canonical_bytes(read32(&slice[pos..])).ok_or(ProofError::FormatError)?;
-        let b = Scalar::from_canonical_bytes(read32(&slice[pos + 32..]))
-            .ok_or(ProofError::FormatError)?;
+        let a = Scalar::from_canonical_bytes(read32(&slice[pos..]));
+        let a = Option::from(a).ok_or(ProofError::FormatError)?;
+        let b = Scalar::from_canonical_bytes(read32(&slice[pos + 32..]));
+        let b = Option::from(b).ok_or(ProofError::FormatError)?;
```

Comment:
This branch uses curve25519-dalek 4.0.0-rc.5, and this [method](https://docs.rs/curve25519-dalek/4.0.0-rc.1/src/curve25519_dalek/scalar.rs.html#262) returns a CtOption instead of an Optoin (like what earlier versions of this method returned). As such, I implemented this (suboptimal) conversion, but am open to other suggestions?

Reply:
Also, we use shadowing in mobilecoin.git normally in cases when you don't want to rename a variable because they are conceptually the same but are of a slightly different type, but point taken.

Reply:
After discussing with @jcape and @nick-mobilecoin, we'll use this conversion technique.